lib/webpush/request.rb

```diff
@@ -36,52 +37,110 @@ def headers
       headers["Content-Type"] = "application/octet-stream"
       headers["Ttl"]          = ttl
 
-      if encrypted_payload?
+      if @payload.has_key?(:server_public_key)
         headers["Content-Encoding"] = "aesgcm"
         headers["Encryption"] = "salt=#{salt_param}"
         headers["Crypto-Key"] = "dh=#{dh_param}"
       end
 
-      headers["Authorization"] = "key=#{api_key}" if api_key?
+      if api_key?
```

Comment:
@rossta Thanks!
I tested your PR and it looks fine.
So I'm gonna merge them if you've got nothing else to check. Is that okay?